client/consensus/slots/src/lib.rs

```diff
@@ -583,9 +597,83 @@ pub fn slot_lenience_linear(parent_slot: u64, slot_info: &SlotInfo) -> Option<Du
 	}
 }
 
+/// Trait for providing the strategy for when to backoff block authoring.
+pub trait BackoffAuthoringBlocksStrategy<N> {
+	/// Returns true if we should backoff authoring new blocks.
+	fn should_backoff(
+		&self,
+		chain_head_number: N,
+		chain_head_slot: u64,
+		finalized_number: N,
+		slow_now: u64,
+	) -> bool;
+}
+
+/// A simple default strategy for how to decide backing off authoring blocks if the number of
+/// unfinalized blocks grows too large.
+#[derive(Clone)]
+pub struct SimpleBackoffAuthoringBlocksStrategy<N> {
+	/// The max interval to backoff when authoring blocks, regardless of delay in finality.
+	pub max_interval: N,
+	/// The number of unfinalized blocks allowed before starting to consider to backoff authoring
+	/// blocks. Note that depending on the value for `authoring_bias`, there might still be an
+	/// additional wait until block authorships starts getting declined.
+	pub unfinalized_slack: N,
+	/// How persistant block authorship is in the face of a growing unfinalized chain of blocks. A
+	/// small value for `authoring_bias` means to quickly start backing off block authorship as the
+	/// length of the unfinalized blocks grows.
```

Comment:
This comment is a bit hard to understand for me, I think it's easier to describe this value as relating to the periodicity at which we author blocks in the face of stalled finality.